open_spiel/games/gin_rummy.h

```diff
@@ -0,0 +1,210 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef THIRD_PARTY_OPEN_SPIEL_GAMES_GIN_RUMMY_H_
+#define THIRD_PARTY_OPEN_SPIEL_GAMES_GIN_RUMMY_H_
+
+// Implementation of the classic card game:
+// https://en.wikipedia.org/wiki/Gin_rummy
+//
+// Gin rummy is played with many variations. Here we closely follow
+// the rules described in http://ginrummytournaments.com/pdfs/GRA_Rules.pdf
+//
+// A game consists of a single hand of gin (i.e. this implementation does not
+// support a full game, consisting of multiple hands, played to some given
+// point total, usually in the 100-300 point range).
+//
+// Gin is a large game with over 10^85 information states and a large number
+// of states per information state. Off the deal there are 41 choose 10 =
+// 1,121,099,408 possible opponent hands, compared to 50 choose 2 = 1,225 in
+// heads up Texas hold 'em poker.
+//
+// Parameters:
+//  "oklahoma"        bool   use oklahoma variation?       (default = false)
+//  "knock_card"      int    set a specific knock card     (default = 10)
+//  "gin_bonus"       int    bonus for getting gin         (default = 25)
+//  "undercut_bonus"  int    bonus for an undercut         (default = 25)
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "open_spiel/spiel.h"
+#include "open_spiel/games/gin_rummy/gin_rummy_utils.h"
+
+namespace open_spiel {
+namespace gin_rummy {
+
+inline constexpr int kInvalidCard = -10000;
+inline constexpr int kNumPlayers = 2;
+inline constexpr int kNumSuits = 4;
+inline constexpr int kNumRanks = 13;
+inline constexpr int kNumCards = kNumSuits * kNumRanks;
+inline constexpr int kMaxPossibleDeadwood = 98;  // E.g. KsKcQdQhJsJcTdTh9s9c
+inline constexpr int kMaxNumDrawUpcardActions = 50;
+inline constexpr int kMaxStockSize = 31;  // Stock size when play begins
+inline constexpr int kWallStockSize = 2;
+inline constexpr int kDefaultKnockCard = 10;
+inline constexpr int kDefaultGinBonus = 25;
+inline constexpr int kDefaultUndercutBonus = 25;
+inline constexpr int kDrawUpcardAction = 52;
+inline constexpr int kDrawStockAction = 53;
+inline constexpr int kPassAction = 54;
+inline constexpr int kKnockAction = 55;
+inline constexpr int kMeldActionBase = 56;  // First lay meld action
+inline constexpr int kNumMeldActions = 185;
+inline constexpr int kNumDistinctActions = kMeldActionBase + kNumMeldActions;
+inline constexpr int kObservationTensorSize =
+    kNumPlayers          // Player turn
+    + kDefaultKnockCard  // Knock card
+    + kNumCards          // Player hand
+    + kNumCards          // Upcard
+    + kNumCards          // Discard pile
+    + kMaxStockSize      // Stock size
+    + kNumMeldActions;   // Opponent's layed melds
+
+class GinRummyGame;
```

Comment:
Tested it and you’re right, it’s not necessary. I had it there because it’s there in a number of other games (leduc_poker, liars_dice, chess, …). Tried removing it on other games and they ran fine too, so it does seem to be superfluous.

Removed in the new commit.